FV3/atmos_cubed_sphere/model/fv_arrays.F90

```diff
@@ -1310,9 +1310,10 @@ module fv_arrays_mod
 ! Others:
 !-----------------------------------------------------------------------
     real, _ALLOCATABLE :: phis(:,:)     _NULL  !< Surface geopotential (g*Z_surf)
-    real, _ALLOCATABLE :: omga(:,:,:)   _NULL  !< Vertical pressure velocity (pa/s)
-    real, _ALLOCATABLE :: ua(:,:,:)     _NULL  !< (ua, va) are mostly used as the A grid winds
-    real, _ALLOCATABLE :: va(:,:,:)     _NULL
+    real, pointer :: omga(:,:,:) => null()
+    ! real, _ALLOCATABLE :: omga(:,:,:)   _NULL  !< Vertical pressure velocity (pa/s)
```

Comment:
One outstanding issue, however, is that in order to be able to point at arrays defined in the `fv_atmos_type` data structure, we must change them from allocatable arrays to pointers [(targets are not allowed in fortran derived types)](https://stackoverflow.com/questions/17114813/using-allocatable-target-variables-in-a-derived-type).  In some compile-time configurations, this is already done -- this is actually what the `_ALLOCATABLE`-`_NULL` business is about; see [this part of `fms_platform.h`](https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/6d852f86d41b44763a92025da2d06771881e8e4b/FMS/include/fms_platform.h#L60-L71) -- but both in FV3GFS and SHiELD by default we compile things in a way that these are defined as allocatable arrays.  I'm guessing this is for a reason.

I am currently discussing with Lucas to see what his preference is -- i.e. is there another possible workaround that would still enable this looping approach, or should we stick with the approach that I originally used in SHiELD, which is verbose, but still allows things be defined using allocatable arrays?

Reply:
It turns out this was a non-issue.  If the derived type is a `target`, then anything inside can be considered a `target` as well.  I just had to make sure I noted `Atm` as a `target` in the definition of the routine I was passing it into.